casper/src/main/scala/coop/rchain/casper/genesis/Genesis.scala

```diff
@@ -103,6 +94,67 @@ object Genesis {
                      )
     } yield genesisBlock
 
+  private def getWallets[F[_]: Sync: Log: RaiseIOError](
+      maybeWalletsPath: Option[String],
+      defaultWalletPath: Path
+  ): F[Seq[Vault]] = {
+    def walletFromFile(walletsPath: Path): F[Seq[Vault]] =
+      for {
+        maybeLines <- SourceIO.open(walletsPath).use(_.getLines).attempt
+        wallets <- maybeLines match {
+                    case Right(lines) =>
+                      lines
+                        .traverse(fromLine(_) match {
+                          case Right(wallet) =>
+                            wallet.some.pure[F]
+                          case Left(errMsg) =>
+                            Sync[F].raiseError[Option[Vault]](
+                              new RuntimeException(s"Error in parsing wallets file: $errMsg")
+                            )
+                        })
+                        .map(_.flatten)
+                    case Left(ex) =>
+                      Sync[F].raiseError(
+                        new RuntimeException(
+                          s"Failed to read ${walletsPath.toAbsolutePath} for reason: ${ex.getMessage}"
+                        )
+                      )
+                  }
+      } yield wallets
+
+    maybeWalletsPath match {
```

Comment:
we should stick with the `vaults` term instead of `wallet`. For future team members it can be confusing to have to use both. Maybe we could also rename `wallets.txt` to `vaults.txt`

Reply:
I agree eventually we should rename the file to vaults.txt but for now let's not surprise the SRE team.